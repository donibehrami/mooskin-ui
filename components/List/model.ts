import { IStackComponentProps } from '../Stack/model';

export interface IListComponentProps extends IStackComponentProps {
    /** number of cols in a list */
    cols?: number;
}
