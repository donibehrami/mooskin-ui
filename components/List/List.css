@import url('https://fonts.googleapis.com/css?family=Convergence');
@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';

.list{
    display: flex;
    flex-direction: column;
    background: #fff;
    border-top: 1px solid #efefef;
    border-bottom: 1px solid #efefef;
}

.listItem{
    display: flex;
    flex-direction: row;
    padding: 25px 0px;
}

.listItem + .listItem {
    border-top: 1px solid #efefef;
}

.image{
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
    overflow: hidden
}

.image img{
    flex-shrink: 0;
    flex-grow: 0; 
    min-width: 100%;
    max-width: 100%;   
}

.titleContainer{
    display: flex;
    flex: 4;
    flex-direction: column;
    justify-content: space-around;
    padding-left: 20px;
}

.detailsContainer{
    display: flex;
    flex: 4;
    flex-direction: row;
}

.title{
    font-family: 'Convergence', sans-serif;
    font-size: 1.25rem;
}

.description{
    color: var(--italicDescription);
    font-size: .78571428571rem;
}

.content {
    flex: 6;
}

.lone{
    flex: 10;
}

.loneImage{
    flex: 1;
}

@media(--xSmall){

    .listItem{
        flex-direction: column;
    }

    .detailsContainer{
        margin-bottom: 15px;
    }

    .titleContainer{
        flex: 3;
    }
}