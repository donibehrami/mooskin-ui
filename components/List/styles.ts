import styled from 'styled-components';

// Models
import { IBoxComponentProps } from '../Box/model';

// Components
import { Box } from '../Box/Box';

// "CSS" variables
import variables from '../_utils/globals/variables';

export const StyledListItem = styled(Box)<IBoxComponentProps>`
    display: flex;
    padding: 15px;
    background-color: ${(props) => props.palette?.layout.backgroundColor || variables.layout.backgroundColor};
    color: ${(props) => props.palette?.layout.fontColor || variables.layout.fontColor};
`;

StyledListItem.displayName = 'StyledListItem';

export const StyledListItemCommon = styled(Box)<IBoxComponentProps>`
    display: flex;
    color: inherit;
`;

StyledListItemCommon.displayName = 'StyledListItemCommon';

export const StyledListItemHead = styled(StyledListItemCommon)``;

StyledListItemHead.displayName = 'StyledListItemHead';

export const StyledListItemBody = styled(StyledListItemCommon)`
    flex: 1;
    flex-direction: column;
    padding: 0 20px;
`;

StyledListItemBody.displayName = 'StyledListItemBody';

export const StyledListItemEnd = styled(StyledListItemCommon)`
    align-items: center;
`;

StyledListItemEnd.displayName = 'StyledListItemEnd';
