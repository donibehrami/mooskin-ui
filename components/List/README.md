# MooSkin UI - List & ListItem Component

The MooSkin List & ListItem Component works similar to the normal HTML `<ol/>`, `<ul/>`, `<li/>` elements, but with pre-defined styling and attributes.

___

### Usage

To start using the List & ListItem Component first you have to Import it

```
Import {List, ListItem, ItemContent} from 'mooskin';
```
or modular import
```
// JS
import List, {ListItem} from 'mooskin/lib/List';

// CSS
import 'mooskin/lib/List/style.css';
```

And then you can simply start using it by typing

```
<List attribute1="atr" attribute2="asd" >
    <ListItem>
        <ItemContent>
            content 1
        <ItemContent/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 2
        <ItemContent/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 3
        <ItemContent/>
    </ListItem>
</List>
```

Like normal html elements, it will accept given attributes and render differently based on the given attributes

### Examples

Give it a custom class or style

```
<List style={yourStyle} className={yourClass} >...</List>
```

or pass props to the LinkItem component

```
<List>
    <ListItem image="imageUrl"/>
    <ListItem title="Some Title"/>
    <ListItem title="Another title" description="Item description" />
</List>
```

### ExpandedSection content

A list item can have an expandable hidden content which will reveal on list item click, example:

```
Import {List, ListItem, ItemContent, ExpandedSection} from 'mooskin';

<List attribute1="atr" attribute2="asd" >
    <ListItem>
        <ItemContent>
            content 1
        <ItemContent/>
        <ExpandedSection>
            expandable content 1
        <ExpandedSection/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 2
        <ItemContent/>
        <ExpandedSection>
            expandable content 2
        <ExpandedSection/>
    </ListItem>
    <ListItem>
        <ItemContent>
            content 3
        <ItemContent/>
        <ExpandedSection>
            expandable content 3
        <ExpandedSection/>
    </ListItem>
</List>
```

<div class="playground-doc">

## Supported List attributes

* `id` - id of the element
* `style` - listitem style
* `className` - css class

## Supported ListItem attributes

* `image` - image related to the list item
* `title` - list item title
* `description` - list item description
* `className` - css class
* `style` - listitem style

</div>

Allthough these attributes are supported, all of them are optional.


#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)