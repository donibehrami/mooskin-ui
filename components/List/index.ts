import List, {ExpandedSection, ItemContent, ListItem} from './List';

export default List;
export {ListItem, ExpandedSection, ItemContent};
