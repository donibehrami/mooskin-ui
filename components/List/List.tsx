import * as React from 'react';

import styles from './List.css';

export interface IListProps {

    /** list id */
    id?: string;

    /** list class */
    className?: string;

    /** color to be applied when an item is hovered */
    hoverColor?: string;

    /** override styles */
    style?: React.CSSProperties;

    /** children here can only be Tab elements */
    children?: React.ReactElement<IListItemProps> | Array<React.ReactElement<IListItemProps>>;

}

export interface IListItemProps {

    /** wether the list item is active to expand inner content */
    active?: boolean;

    /** listitem image */
    image?: string;

    /** listitem title */
    title?: string;

    /** listitem description */
    description?: string;

    /** inherited from parent */
    hovered?: boolean;

    /** color to be applied when this item is hovered (inherited by default) */
    hoverColor?: string;

    /** listitem class */
    className?: string;

    /** override styles */
    style?: React.CSSProperties;

    /** listitem children */
    children?: React.ReactElement<IExpandedSectionProps> | React.ReactElement<IItemContentProps> |
        Array<React.ReactElement<IExpandedSectionProps> | React.ReactElement<IItemContentProps>>;
    // children?: any;

    onMouseEnter?: (e: React.MouseEvent<HTMLElement>) => void;

    onMouseLeave?: (e: React.MouseEvent<HTMLElement>) => void;

    onClick?: (e: React.MouseEvent<HTMLElement>) => void;
}

export interface IListState {
    hoverItem?: number;
    activeLists: number[];
}

export default class List extends React.Component<IListProps, IListState>{

    static defaultProps = {
        className: '',
        style: {}
    };

    constructor(props: IListProps){
        super(props);

        this.state = {
            activeLists: [],
            hoverItem: -1
        };
    }

    render(){

        const classes = `${styles.list} ${this.props.className}`;

        return(
            <div className={`list-component ${classes}`} style={this.props.style} id={this.props.id}>
                {this.getItems()}
            </div>
        );
    }

    getItems = () => {
        const items: Array<React.ReactElement<IListItemProps>> = [];
        React.Children.forEach(this.props.children, (child, index) => {
            if (React.isValidElement<IListItemProps>(child)){
                const {content, expandedSection} = child.props.children ?
                    this.getItemChildren(child.props.children, index) : {content: null, expandedSection: null};
                const onClick = expandedSection ? this.onListClick(index) : undefined;
                child.props.active && this.setActiveState(index);
                const active = child.props.active ? child.props.active
                            : this.state.activeLists.includes(index) ? true : false;
                const cursor = expandedSection ? {cursor: 'pointer'} : {};
                items.push(
                    <div style={{display: 'flex', flexDirection: 'column'}} key={index}>
                        <ListItem
                            active={active}
                            hovered={index === this.state.hoverItem}
                            className={child.props.className}
                            image={child.props.image}
                            description={child.props.description}
                            title={child.props.title}
                            style={{...cursor, ...child.props.style}}
                            hoverColor={child.props.hoverColor || this.props.hoverColor}
                            onMouseEnter={this.onItemEnter(index)}
                            onMouseLeave={this.onItemLeave}
                            onClick={onClick}
                        >
                            {content}
                        </ListItem>
                        {expandedSection}
                    </div>
                );
            }
        });
        return items;
    }

    onListClick = (index: number) => {
        return (e: React.MouseEvent<HTMLElement>) => {
            e.stopPropagation();
            if (this.state.activeLists.includes(index)){
                const activeLists = [...this.state.activeLists];
                const i = activeLists.indexOf(index);
                activeLists.splice(i, 1);
                this.setState({activeLists});
            } else {
                const activeLists = [...this.state.activeLists];
                activeLists.push(index);
                this.setState({activeLists});
            }
        };
    }

    setActiveState = (index: number) => {
        if (!this.state.activeLists.includes(index)){
            const activeLists = [...this.state.activeLists];
            activeLists.push(index);
            this.setState({activeLists});
        }
    }

    onItemEnter = (index: number) => {
        return (e: React.MouseEvent<HTMLElement>) => {
            this.setState({hoverItem: index});
        };
    }

    onItemLeave = () => {
        this.setState({hoverItem: -1});
    }

    getItemChildren = (children: any, index: number) => {
        let content;
        let expandedSection;
        if (Array.isArray(children)){
            children.forEach((child, i) => {
                if (child.type === ExpandedSection){
                    const active = this.state.activeLists.includes(index) ? true : false;
                    expandedSection = (
                        <ExpandedSection active={active}>
                            {child.props.children}
                        </ExpandedSection>
                    );
                } else {
                    content = child;
                }
            });
        } else {
            if (children.type === ExpandedSection){
                const active = this.state.activeLists.includes(index) ? true : false;
                expandedSection = (
                    <ExpandedSection active={active}>
                        {children.props.children}
                    </ExpandedSection>
                );
            } else {
                content = children;
            }
        }
        return {content, expandedSection};
    }
}

export const ListItem: React.StatelessComponent<IListItemProps> = (props) => {

    const {className, description, image, style, title} = props;

    const hoverColor = props.hoverColor && props.hovered ? {backgroundColor: props.hoverColor} : {};

    const getImage = () => {
        return (
            <div className={styles.image}>
                <img src={image}/>
            </div>
        );
    };

    const getTitle = () => {
        return title || description ?
        (
            <div className={`${styles.titleContainer}`}>
                {title && <span className={styles.title}>{title}</span>}
                {description && <span className={styles.description}>{description}</span>}
            </div>
        ) : null;
    };

    const getContent = () => {
        const classes = title || description ? '' : styles.lone;
        return props.children &&
        (
            <div className={`${styles.content} ${classes}`}>
                {props.children}
            </div>
        );
    };

    const getDetails = () => {
        const classes = !title || !description ? styles.loneImage : '';
        return image || title || description ?
        (
            <div className={`${styles.detailsContainer} ${classes}`}>
                {image && getImage()}
                {getTitle()}
            </div>
        ) : null;
    };

    const arrow = () => {
        return props.active ? (
            <div className={styles.arrow}>
                <div style={{position: 'relative', height: '100%', width: '100%'}}>
                    <div className={styles.innerArrow}/>
                    <div className={styles.borderArrow}/>
                </div>
            </div>
        ) : null;
    };

    return(
        <div
            className={`listitem-component ${styles.listItem} ${className}`}
            style={{...style, ...hoverColor}}
            onMouseEnter={props.onMouseEnter}
            onMouseLeave={props.onMouseLeave}
            onClick={props.onClick}
        >
            {arrow()}
            {getDetails()}
            {getContent()}
        </div>
    );
};

export interface IExpandedSectionProps {
    className?: string;
    style?: React.CSSProperties;
    active?: boolean;
}

export const ExpandedSection: React.StatelessComponent<IExpandedSectionProps> = (props) => {

    const display = props.active ? {display: 'block'} : {display: 'none'};

    return(
        <div
            className={`expanded-section-component ${styles.expandedSection} ${props.className}`}
            style={{...display, ...props.style}}
        >
            {props.children}
        </div>
    );
};

export interface IItemContentProps {
    className?: string;
    style?: React.CSSProperties;
}

export const ItemContent: React.StatelessComponent<IItemContentProps> = (props) => {

    return(
        <div
            className={`item-content-component ${props.className}`}
            style={props.style}
        >
            {props.children}
        </div>
    );
};

ListItem.defaultProps = {
    className: '',
    style: {}
};
