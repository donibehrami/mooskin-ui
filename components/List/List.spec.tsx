import * as React from 'react';
import List, {ExpandedSection, ItemContent, ListItem} from './List';

import { mount, shallow } from 'enzyme';

describe('Button', () => {

    test('renders List correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            />
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders ListItem correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <ListItem
                description="desc"
                title="title"
                className="myClass"
                style={{color: 'blue'}}
                image="imageUrl"
            >
                <ItemContent>
                    test
                </ItemContent>
            </ListItem>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders with multiple children', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    description="desc"
                    title="title"
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        test
                    </ItemContent>
                </ListItem>
                <ListItem
                    className="myClass"
                    style={{color: 'blue'}}
                    image="imageUrl"
                >
                    <ItemContent>
                        test
                    </ItemContent>
                </ListItem>
            </List>
        );

        expect(component).toMatchSnapshot();
    });

    test('renders ListItems correctly depending on props', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    description="desc"
                    title="title"
                    className="myClass"
                    style={{color: 'blue'}}
                />
            </List>
        );

        expect(component.find('.detailsContainer').length).toBe(1);

        expect(component.find('img').length).toBe(0);
        expect(component.find('.content').length).toBe(0);

        expect(component.find('.title').length).toBe(1);
        expect(component.find('.description').length).toBe(1);

    });

    test('renders ListItems correctly with different props', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    image="imageURL"
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        Some Content
                    </ItemContent>
                </ListItem>
            </List>
        );

        expect(component.find('.detailsContainer').length).toBe(1);

        expect(component.find('img').length).toBe(1);
        expect(component.find('.content').length).toBe(1);

        expect(component.find('.title').length).toBe(0);
        expect(component.find('.description').length).toBe(0);

    });

    test('applies the correct classes depending on ListItem content', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    image="imageURL"
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        Some Content
                    </ItemContent>
                </ListItem>
            </List>
        );

        expect(component.find('.detailsContainer').hasClass('loneImage')).toBe(true);
        expect(component.find('.content').hasClass('lone')).toBe(true);

    });

    test('create expanded section content correctly', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    image="imageURL"
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection>
                        Some ExpandedSection Content
                    </ExpandedSection>
                </ListItem>
            </List>
        );

        expect(component).toMatchSnapshot();
    });

    test('create expanded section content when proper child is assigned', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem
                    image="imageURL"
                    className="myClass"
                    style={{color: 'blue'}}
                >
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection>
                        Some ExpandedSection Content
                    </ExpandedSection>
                </ListItem>
            </List>
        );

        expect(component.find('.expanded-section-component').length).toBe(1);
        expect(component.find('.expanded-section-component').prop('style')).toEqual({display: 'none'});
        expect(component.state('activeLists')).toEqual([]);

        component.find('.listitem-component').simulate('click');

        expect(component.find('.expanded-section-component').prop('style')).toEqual({display: 'block'});
        expect(component.state('activeLists')).toEqual([0]);

    });

    test('render multiple expanded section items and multi items can be active', () => {

        const component = mount(
            <List
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
            >
                <ListItem>
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection>
                        Some ExpandedSection Content
                    </ExpandedSection>
                </ListItem>
                <ListItem>
                    <ItemContent>
                        Some Content
                    </ItemContent>
                    <ExpandedSection>
                        ExpandedSection content the second
                    </ExpandedSection>
                </ListItem>
            </List>
        );

        expect(component.find('.expanded-section-component').length).toBe(2);
        expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'none'});
        expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});
        expect(component.state('activeLists')).toEqual([]);

        component.find('.listitem-component').first().simulate('click');

        expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'block'});
        expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});
        expect(component.state('activeLists')).toEqual([0]);

        component.find('.listitem-component').last().simulate('click');

        expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'block'});
        expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'block'});
        expect(component.state('activeLists')).toEqual([0, 1]);

        component.find('.listitem-component').first().simulate('click');

        expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'none'});
        expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'block'});
        expect(component.state('activeLists')).toEqual([1]);

        component.find('.listitem-component').last().simulate('click');

        expect(component.find('.expanded-section-component').first().prop('style')).toEqual({display: 'none'});
        expect(component.find('.expanded-section-component').last().prop('style')).toEqual({display: 'none'});
        expect(component.state('activeLists')).toEqual([]);

    });

});
