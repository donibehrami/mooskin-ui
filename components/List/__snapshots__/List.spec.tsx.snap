// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button create expanded section content correctly 1`] = `
<List
  className="myClass"
  id="button1"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="list-component list myClass"
    id="button1"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <div
      key="0"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <ListItem
        active={false}
        className="myClass"
        hovered={false}
        image="imageURL"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "blue",
            "cursor": "pointer",
          }
        }
      >
        <div
          className="listitem-component listItem myClass"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "blue",
              "cursor": "pointer",
            }
          }
        >
          <div
            className="detailsContainer loneImage"
          >
            <div
              className="image"
            >
              <img
                src="imageURL"
              />
            </div>
          </div>
          <div
            className="content lone"
          >
            <ItemContent>
              <div
                className="item-content-component undefined"
              >
                Some Content
              </div>
            </ItemContent>
          </div>
        </div>
      </ListItem>
      <ExpandedSection
        active={false}
      >
        <div
          className="expanded-section-component expandedSection undefined"
          style={
            Object {
              "display": "none",
            }
          }
        >
          Some ExpandedSection Content
        </div>
      </ExpandedSection>
    </div>
  </div>
</List>
`;

exports[`Button renders List correctly 1`] = `
<div
  className="list-component list myClass"
  id="button1"
  style={
    Object {
      "color": "blue",
    }
  }
/>
`;

exports[`Button renders ListItem correctly 1`] = `
<div
  className="listitem-component listItem myClass"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="detailsContainer "
  >
    <div
      className="image"
    >
      <img
        src="imageUrl"
      />
    </div>
    <div
      className="titleContainer"
    >
      <span
        className="title"
      >
        title
      </span>
      <span
        className="description"
      >
        desc
      </span>
    </div>
  </div>
  <div
    className="content "
  >
    <ItemContent>
      test
    </ItemContent>
  </div>
</div>
`;

exports[`Button renders with multiple children 1`] = `
<List
  className="myClass"
  id="button1"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <div
    className="list-component list myClass"
    id="button1"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <div
      key="0"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <ListItem
        active={false}
        className="myClass"
        description="desc"
        hovered={false}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
        title="title"
      >
        <div
          className="listitem-component listItem myClass"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <div
            className="detailsContainer "
          >
            <div
              className="titleContainer"
            >
              <span
                className="title"
              >
                title
              </span>
              <span
                className="description"
              >
                desc
              </span>
            </div>
          </div>
          <div
            className="content "
          >
            <ItemContent>
              <div
                className="item-content-component undefined"
              >
                test
              </div>
            </ItemContent>
          </div>
        </div>
      </ListItem>
    </div>
    <div
      key="1"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <ListItem
        active={false}
        className="myClass"
        hovered={false}
        image="imageUrl"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <div
          className="listitem-component listItem myClass"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <div
            className="detailsContainer loneImage"
          >
            <div
              className="image"
            >
              <img
                src="imageUrl"
              />
            </div>
          </div>
          <div
            className="content lone"
          >
            <ItemContent>
              <div
                className="item-content-component undefined"
              >
                test
              </div>
            </ItemContent>
          </div>
        </div>
      </ListItem>
    </div>
  </div>
</List>
`;
